pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PicklistAnswerDao.java

```diff
@@ -70,7 +73,14 @@ default void assignOptionsToAnswerId(long answerId, List<SelectedPicklistOption>
 
         List<Long> selectedIds = new ArrayList<>();
         List<String> detailTexts = new ArrayList<>();
-        for (SelectedPicklistOption option : selected) {
+        // sort options by their database id for a particular question and ensure inserted in same
+        // order as database index and minimize deadlock opportunity
+        Function<SelectedPicklistOption, Long> getOptionDbId = (o) -> Optional.ofNullable(dtoMap.get(o.getStableId()))
+                        .map(PicklistOptionDto::getId).orElse(Long.MAX_VALUE);
+        List<SelectedPicklistOption> sortedSelected = new ArrayList(selected);
+        sortedSelected.sort(Comparator.comparingLong(getOptionDbId::apply));
```

Comment:
:+1:

Reply:
We make a patch call for each option selected. Ideally we should see only 1 insert in each patch call but during batching , sorting should help.